logstash-core/lib/logstash/environment.rb

```diff
@@ -109,7 +109,8 @@ module Environment
          Setting::TimeValue.new("slowlog.threshold.trace", "-1"),
             Setting::String.new("keystore.classname", "org.logstash.secret.store.backend.JavaKeyStore"),
             Setting::String.new("keystore.file", ::File.join(::File.join(LogStash::Environment::LOGSTASH_HOME, "config"), "logstash.keystore"), false), # will be populated on
-    Setting::NullableString.new("monitoring.cluster_uuid")
+    Setting::NullableString.new("monitoring.cluster_uuid"),
+            Setting::String.new("pipeline.receive_buffer.type", "direct", true, ["direct", "heap"])
```

Comment:
I'd generalize to "buffer" instead of just "receive" buffer:

```suggestion
            Setting::String.new("pipeline.buffer.type", "direct", true, ["direct", "heap"])
```